diff --git a/.cargo/config.toml b/.cargo/config.toml
index f0ba8af..5e5b3a0 100644
--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -13,3 +13,8 @@ rustflags = [
   "-C", "link-arg=-undefined",
   "-C", "link-arg=dynamic_lookup",
 ]
+
+[patch.crates-io]
+speedate = { git = 'https://github.com/DataDog/speedate.git', branch = 'support-old-nightly' }
+pyo3 = { git = 'https://github.com/DataDog/pyo3.git', branch = 'support-old-nightly' }
+hashbrown = { git = 'https://github.com/DataDog/hashbrown.git', branch = 'support-old-nightly' }
diff --git a/Cargo.toml b/Cargo.toml
index 1cfdba5..058325c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -52,7 +52,6 @@ crate-type = ["cdylib", "rlib"]
 [features]
 # must be enabled when building with `cargo build`, maturin enables this automatically
 extension-module = ["pyo3/extension-module"]
-default = ["mimalloc"]
 
 [profile.release]
 lto = "fat"
diff --git a/pyproject.toml b/pyproject.toml
index 1602e32..0278c38 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [build-system]
 requires = [
-    'maturin>=1,<2',
+    'maturin[zig]>=1,<2',
     'typing-extensions >=4.6.0,!=4.7.0'
 ]
 build-backend = 'maturin'
@@ -52,6 +52,7 @@ python-source = "python"
 module-name = "pydantic_core._pydantic_core"
 bindings = 'pyo3'
 features = ["pyo3/extension-module"]
+compatibility = "manylinux1"
 
 [tool.ruff]
 line-length = 120
