--- a/internal.bzl	2025-10-31 16:00:48.767972855 +0100
+++ b/internal.bzl	2025-11-18 10:16:13.143118991 +0100
@@ -42,7 +42,7 @@
 def _avl():
     http_archive(
         name = "avl",
-        build_file = "@//:examples/avl/avl.BUILD.bazel",
+        build_file = "//:examples/avl/avl.BUILD.bazel",
         sha256 = "6d62e563578b79795a84958cfe4e221a4c9847fbeb4a821d45bc049934fc6a90",
         strip_prefix = "Avl",
         url = "https://web.mit.edu/drela/Public/web/avl/avl3.40b.tgz",
