From a9e4057bfda190ad365b503af058b460ab8c7616 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Tue, 1 Oct 2013 22:22:57 +0200
Subject: [PATCH] aclocal: Add parameter to disable keyutils detection

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>

---
 aclocal.m4 | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/aclocal.m4 b/aclocal.m4
index d6d1279..80ce604 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -1679,12 +1679,16 @@ fi
 dnl
 dnl If libkeyutils exists (on Linux) include it and use keyring ccache
 AC_DEFUN(KRB5_AC_KEYRING_CCACHE,[
+AC_ARG_ENABLE([keyutils],
+AC_HELP_STRING([--disable-keyutils],don't enable using keyutils for keyring ccache @<:@enabled@:>@), , enable_keyutils=yes)
+if test "$enable_keyutils" = yes; then
   AC_CHECK_HEADERS([keyutils.h],
     AC_CHECK_LIB(keyutils, add_key, 
       [dnl Pre-reqs were found
        AC_DEFINE(USE_KEYRING_CCACHE, 1, [Define if the keyring ccache should be enabled])
        LIBS="-lkeyutils $LIBS"
       ]))
+fi
 ])dnl
 dnl
 dnl If libkeyutils supports persistent keyrings, use them
