From 3b255c93a6462b4a802801a9155e140ac307166f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo.beauzee@datadoghq.com>
Date: Mon, 19 May 2025 14:28:36 +0200
Subject: [PATCH] disable doc & tests

---
 Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 4c4deba6..1416ca10 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,11 +1,11 @@
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = libxslt libexslt xsltproc doc tests
+SUBDIRS = libxslt libexslt xsltproc
 if WITH_PYTHON
 SUBDIRS += python
 endif
 
-DIST_SUBDIRS = libxslt libexslt xsltproc python doc tests
+DIST_SUBDIRS = libxslt libexslt xsltproc python
 
 confexecdir=$(libdir)
 confexec_DATA = xsltConf.sh
-- 
2.34.1

