From 811a79ef3a44e72e6535b782ead13649cb863c7e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo.beauzee@datadoghq.com>
Date: Mon, 17 Feb 2025 12:05:06 +0100
Subject: [PATCH] add missing include dirs

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fdeda6eb4..ebb5a9011 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -582,6 +582,9 @@ include_directories(
 	${OPENSSL_INCLUDE_DIR}
 	${PCRE_INCLUDE_DIRS}
 	${PCRE2_INCLUDE_DIRS}
+    ${LIBXSLT_INCLUDE_DIRS}
+    ${RPM_INCLUDE_DIRS}
+    ${GCRYPT_INCLUDE_DIRS}
 )
 
 # Honor visibility properties for all target types
-- 
2.34.1

