From e39816e71f5aa1535d18e38bf51e2789501edc90 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo.beauzee@datadoghq.com>
Date: Mon, 17 Feb 2025 12:05:06 +0100
Subject: [PATCH 1/2] add missing include dirs

---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fdeda6eb4..80b936e9c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -582,6 +582,12 @@ include_directories(
 	${OPENSSL_INCLUDE_DIR}
 	${PCRE_INCLUDE_DIRS}
 	${PCRE2_INCLUDE_DIRS}
+    ${LIBXSLT_INCLUDE_DIR}
+    ${RPM_INCLUDE_DIRS}
+    ${GCRYPT_INCLUDE_DIRS}
+    ${BLKID_INCLUDE_DIRS}
+    ${POPT_INCLUDE_DIRS}
+    ${SELINUX_INCLUDE_DIRS}
 )
 
 # Honor visibility properties for all target types
-- 
2.34.1

