diff --git a/src/OVAL/probes/fsdev.c b/src/OVAL/probes/fsdev.c
index 05db635cf..21eadf508 100644
--- a/src/OVAL/probes/fsdev.c
+++ b/src/OVAL/probes/fsdev.c
@@ -66,6 +66,7 @@
 
 #include "fsdev.h"
 #include "common/util.h"
+#include "debug_priv.h"
 
 /**
  * Compare two dev_t variables.
@@ -110,6 +111,7 @@ static int is_local_fs(struct mntent *ment)
 	const char *pseudo_fs[] = {
 		"proc",
 		"sysfs",
+		"overlay",
 		NULL
 	};
 	const char *network_fs[] = {
@@ -141,6 +143,15 @@ static int is_local_fs(struct mntent *ment)
 		if (!strcmp(network_fs[i], fstype))
 			return 0;
 	}
+	// ignore /host directory
+	const char* host = "host";
+	if (oscap_str_startswith(ment->mnt_dir, host)) {
+		const char* root = getenv("OSCAP_PROBE_ROOT");
+		if ((root == NULL) || (strcmp(root, host) != 0)) {
+			dW("Skipping '%s' directory.", ment->mnt_dir);
+			return 0;
+		}
+	}
 	return 1;
 }
 
