--- a/src/pip/_internal/cli/base_command.py
+++ b/src/pip/_internal/cli/base_command.py
@@ -145,17 +145,6 @@ class Command(object):
                 replacement=None,
                 gone_in='19.2',
             )
-        elif sys.version_info[:2] == (2, 7):
-            message = (
-                "A future version of pip will drop support for Python 2.7."
-            )
-            if platform.python_implementation() == "CPython":
-                message = (
-                    "Python 2.7 will reach the end of its life on January "
-                    "1st, 2020. Please upgrade your Python as Python 2.7 "
-                    "won't be maintained after that date. "
-                ) + message
-            deprecated(message, replacement=None, gone_in=None)

         # TODO: Try to get these passing down from the command?
         #       without resorting to os.environ to hold these.
