diff --git a/buildifier/BUILD.bazel b/buildifier/BUILD.bazel
index 1f1cd67..2f2a5b1 100644
--- a/buildifier/BUILD.bazel
+++ b/buildifier/BUILD.bazel
@@ -1,3 +1,4 @@
+INHIBITED_IN_FAVOR_OF_PREBUILT = """
 load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
 load("@rules_shell//shell:sh_test.bzl", "sh_test")
 
@@ -108,6 +109,7 @@ go_library(
         "//wspace",
     ],
 )
+"""
 
 exports_files(
     [
