From 8e4cf56287603160f66e982b798eb7d13c1367c0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo.beauzee@datadoghq.com>
Date: Mon, 9 Mar 2026 15:04:08 +0100
Subject: [PATCH] buildifier: add *.BUILD.bazel to the list of analyzed
 patterns

---
 buildifier/runner.bash.template | 1 +
 1 file changed, 1 insertion(+)

diff --git a/buildifier/runner.bash.template b/buildifier/runner.bash.template
index 06a969b..32f00f6 100644
--- a/buildifier/runner.bash.template
+++ b/buildifier/runner.bash.template
@@ -50,6 +50,7 @@ find . \
     -o -name BUILD.bazel \
     -o -name BUILD \
     -o -name '*.BUILD' \
+    -o -name '*.BUILD.bazel' \
     -o -name 'BUILD.*.bazel' \
     -o -name 'BUILD.*.oss' \
     -o -name MODULE.bazel \
-- 
2.43.0

