diff --git a/buildifier/runner.bat.template b/buildifier/runner.bat.template
index 13ca3fa..d8ccac9 100644
--- a/buildifier/runner.bat.template
+++ b/buildifier/runner.bat.template
@@ -11,9 +11,34 @@ set stripped_args=%stripped_args:'=%
 rem Get the absolute path to the buildifier executable
-for /f "tokens=2" %%i in ('findstr /r "\<buildifier\.exe\>" MANIFEST') do (set buildifier_abs_path=%%i)
+if exist MANIFEST (
+    set manifest_file=MANIFEST
+) else if exist "%RUNFILES_MANIFEST_FILE%" (
+    set "manifest_file=%RUNFILES_MANIFEST_FILE:/=\%"
+) else (
+    >&2 echo Error: manifest file not found
+    exit /b 1
+)
+for /f "tokens=2" %%i in ('findstr /r "\<buildifier\.exe\>" "%manifest_file%"') do (set buildifier_abs_path=%%i)
 
 powershell ^
+function Should-Exclude($Path)^
+{^
+    $relPath = '.' + $Path.Substring('%BUILD_WORKSPACE_DIRECTORY%'.Length);^
+    foreach ($pattern in @(@@EXCLUDE_PATTERNS@@))^
+    {^
+        if ($relPath -clike $pattern)^
+        {^
+            return $true;^
+        };^
+    };^
+    return $false;^
+};^
 function Buildify($Root)^
 {^
+    if (Should-Exclude $Root)^
+    {^
+        return;^
+    };^
     $Folder = (New-Object -Com Scripting.FileSystemObject).GetFolder($Root);^
     $Files = $Folder.Files ^| Where-Object {^
+        (^
         $_.Name -eq 'BUILD.bazel' `^
@@ -30,2 +55,3 @@ function Buildify($Root)^
         -or $_.Name -clike 'WORKSPACE.*.oss'^
+        ) -and -not (Should-Exclude $_.Path)^
     };^
@@ -44,2 +70,2 @@ function Buildify($Root)^
 };^
-Buildify('%BUILD_WORKSPACE_DIRECTORY%');
+Buildify('%BUILD_WORKSPACE_DIRECTORY:/=\%');
