--- a/BUILD.bazel	2026-06-10 02:15:05.114430587 +0000
+++ b/BUILD.bazel	2026-06-10 02:15:21.502065584 +0000
@@ -1,4 +1,5 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@rules_cc//cc:defs.bzl", "cc_library")
 
 # gazelle:proto disable
 # gazelle:resolve go github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace @@//pkg/proto/pbgo/trace:trace
@@ -34,6 +35,24 @@
 # gazelle:resolve go github.com/DataDog/datadog-agent/pkg/remoteconfig/state @@//pkg/remoteconfig/state:state
 # gazelle:resolve go github.com/DataDog/datadog-agent/pkg/fleet/installer/telemetry @@//pkg/fleet/installer/telemetry:telemetry
 
+cc_library(
+    name = "odpi",
+    hdrs = glob([
+        "odpi/include/*.h",
+        "odpi/src/*.h",
+    ]),
+    includes = [
+        "odpi/embed",
+        "odpi/include",
+        "odpi/src",
+    ],
+    textual_hdrs = glob([
+        "odpi/embed/dpi.c",
+        "odpi/src/*.c",
+    ]),
+    visibility = ["//visibility:public"],
+)
+
 go_library(
     name = "godror",
     srcs = [
@@ -61,6 +80,7 @@
         "vector.go",
         "version.go",
     ],
+    cdeps = [":odpi"],
     cgo = True,
     clinkopts = select({
         "@io_bazel_rules_go//go/platform:android": [
