--- a/pcap/BUILD.bazel	2026-06-09 22:15:20.567736417 +0000
+++ b/pcap/BUILD.bazel	2026-06-09 22:16:35.327085662 +0000
@@ -11,9 +11,13 @@
         "pcap_windows.go",
     ],
     cgo = True,
+    cdeps = select({
+        "@io_bazel_rules_go//go/platform:android": ["@@+http_archive+libpcap//:pcap"],
+        "@io_bazel_rules_go//go/platform:linux": ["@@+http_archive+libpcap//:pcap"],
+        "//conditions:default": [],
+    }),
     clinkopts = select({
         "@io_bazel_rules_go//go/platform:android": [
-            "-lpcap",
         ],
         "@io_bazel_rules_go//go/platform:darwin": [
             "-lpcap",
@@ -28,7 +32,6 @@
             "-lpcap",
         ],
         "@io_bazel_rules_go//go/platform:linux": [
-            "-lpcap",
         ],
         "@io_bazel_rules_go//go/platform:netbsd": [
             "-lpcap",
