--- a/pcap/BUILD.bazel
+++ b/pcap/BUILD.bazel
@@ -10,6 +10,10 @@
         "pcap_unix.go",
         "pcap_windows.go",
     ],
+    cdeps = select({
+        "@io_bazel_rules_go//go/platform:linux": ["@libpcap//:pcap"],
+        "//conditions:default": [],
+    }),  # keep
     cgo = True,
     clinkopts = select({
         "@io_bazel_rules_go//go/platform:android": [
@@ -27,9 +31,6 @@
         "@io_bazel_rules_go//go/platform:ios": [
             "-lpcap",
         ],
-        "@io_bazel_rules_go//go/platform:linux": [
-            "-lpcap",
-        ],
         "@io_bazel_rules_go//go/platform:netbsd": [
             "-lpcap",
         ],
