diff --git a/runtimes/mingw/BUILD.bazel b/runtimes/mingw/BUILD.bazel
--- a/runtimes/mingw/BUILD.bazel
+++ b/runtimes/mingw/BUILD.bazel
@@ -330,9 +330,19 @@
     name = "stdc++",
 )
 
+stub_library(
+    name = "gcc",
+)
+
+stub_library(
+    name = "gcc_eh",
+)
+
 copy_to_directory(
     name = "mingw_crt_library_search_directory",
     srcs = [
+        ":gcc",
+        ":gcc_eh",
         ":m",
         ":mingw32",
         ":mingwex",
diff --git a/toolchain/llvm/llvm.bzl b/toolchain/llvm/llvm.bzl
--- a/toolchain/llvm/llvm.bzl
+++ b/toolchain/llvm/llvm.bzl
@@ -294,6 +294,7 @@
         src = "bin/clang++" + suffix,
         data = [
             "bin/ld.lld" + suffix,
+            "bin/llvm-dlltool" + suffix,
             "bin/ld64.lld" + suffix,
             "bin/lld" + suffix,
             "bin/wasm-ld" + suffix,
