diff --git a/examples/rpm/debuginfo/README.md b/examples/rpm/debuginfo/README.md
index 08f8957..e552f8c 100644
--- a/examples/rpm/debuginfo/README.md
+++ b/examples/rpm/debuginfo/README.md
@@ -6,7 +6,7 @@ This example uses the `find_system_rpmbuild_bzlmod` module extension to help
 us register the system rpmbuild as a toolchain in a bzlmod environment.
 
 It configures the system toolchain to be aware of which debuginfo configuration
-to use (defaults to "none", the example uses "centos" for RPM < 4.18).
+to use (defaults to "none", the example uses "centos" for RPM < 4.18 or >= 4.20).
 
 ## To use
 
diff --git a/pkg/legacy/tests/rpm/test_rpm.spec b/pkg/legacy/tests/rpm/test_rpm.spec
index efaeb6f..fc255b0 100644
--- a/pkg/legacy/tests/rpm/test_rpm.spec
+++ b/pkg/legacy/tests/rpm/test_rpm.spec
@@ -18,7 +18,7 @@ This is a package description.
 %build
 
 %install
-cp -r ./pkg/legacy %{buildroot}/
+cp -r %{_topdir}/BUILD/pkg/legacy %{buildroot}/
 
 %files
 /legacy/rpm.bzl
diff --git a/pkg/make_rpm.py b/pkg/make_rpm.py
index a6a84b8..4835624 100644
--- a/pkg/make_rpm.py
+++ b/pkg/make_rpm.py
@@ -410,12 +410,8 @@ class RpmBuilder(object):
     if self.install_script_file:
       args += ['--define', 'build_rpm_install %s' % self.install_script_file]
     if self.file_list_path:
-      # %files -f is taken relative to the package root
-      base_path = os.path.basename(self.file_list_path)
-      if debuginfo_type == RpmBuilder.DEBUGINFO_TYPE_FEDORA:
-        base_path = os.path.join("..", base_path)
-
-      args += ['--define', 'build_rpm_files %s' % base_path]
+      # %files -f is taken relative to the build directory, which varies across RPM versions
+      args += ['--define', 'build_rpm_files %s' % os.path.join(dirname, self.file_list_path)]
 
     args.extend(rpmbuild_args)
 
diff --git a/pkg/rpm/augment_rpm_files_install.py b/pkg/rpm/augment_rpm_files_install.py
index 6f0762b..13855b4 100644
--- a/pkg/rpm/augment_rpm_files_install.py
+++ b/pkg/rpm/augment_rpm_files_install.py
@@ -25,7 +25,7 @@ import json
 # NOTE: Keep those two in sync with the same variables in rpm_pfg.bzl
 _INSTALL_FILE_STANZA_FMT = """
 install -d "%{{buildroot}}/$(dirname '{1}')"
-cp '{0}' '%{{buildroot}}/{1}'
+cp '%{{_topdir}}/BUILD/{0}' '%{{buildroot}}/{1}'
 """.strip()
 
 _FILE_MODE_STANZA_FMT = """
diff --git a/pkg/rpm_pfg.bzl b/pkg/rpm_pfg.bzl
index 4332a59..3f66dde 100644
--- a/pkg/rpm_pfg.bzl
+++ b/pkg/rpm_pfg.bzl
@@ -34,11 +34,7 @@ load(
     "PackageVariablesInfo",
 )
 load("//pkg/private:util.bzl", "get_stamp_detect", "setup_output_files", "substitute_package_variables")
-load(
-    "//toolchains/rpm:rpmbuild_configure.bzl",
-    "DEBUGINFO_TYPE_FEDORA",
-    "DEBUGINFO_TYPE_NONE",
-)
+load("//toolchains/rpm:rpmbuild_configure.bzl", "DEBUGINFO_TYPE_NONE")
 
 rpm_filetype = [".rpm"]
 
@@ -73,13 +69,7 @@ DEFAULT_FILE_MODE = "%defattr(-,root,root)"
 # TODO(nacl, #292): cp -r does not do the right thing with TreeArtifacts
 _INSTALL_FILE_STANZA_FMT = """
 install -d "%{{buildroot}}/$(dirname '{1}')"
-cp '{0}' '%{{buildroot}}/{1}'
-chmod +w '%{{buildroot}}/{1}'
-""".strip()
-
-_INSTALL_FILE_STANZA_FMT_FEDORA40_DEBUGINFO = """
-install -d "%{{buildroot}}/$(dirname '{1}')"
-cp '../{0}' '%{{buildroot}}/{1}'
+cp '%{{_topdir}}/BUILD/{0}' '%{{buildroot}}/{1}'
 chmod +w '%{{buildroot}}/{1}'
 """.strip()
 
@@ -202,7 +192,7 @@ def _make_rpm_filename(rpm_name, version, architecture, package_name = None, rel
 # TODO(nacl, #459): These are redundant with functions and structures in
 # pkg/private/pkg_files.bzl.  We should really use the infrastructure provided
 # there, but as of writing, it's not quite ready.
-def _process_files(pfi, origin_label, grouping_label, file_base, rpm_ctx, debuginfo_type):
+def _process_files(pfi, origin_label, grouping_label, file_base, rpm_ctx):
     for dest, src in pfi.dest_src_map.items():
         metadata = _package_contents_metadata(origin_label, grouping_label)
         abs_dest = _make_absolute_if_not_already_or_is_macro(dest)
@@ -226,12 +216,7 @@ def _process_files(pfi, origin_label, grouping_label, file_base, rpm_ctx, debugi
 
             # Files are well-known.  Take care of them right here.
             rpm_ctx.rpm_files_list.append(_FILE_MODE_STANZA_FMT.format(file_base, abs_dest))
-
-            install_stanza_fmt = _INSTALL_FILE_STANZA_FMT
-            if debuginfo_type == DEBUGINFO_TYPE_FEDORA:
-                install_stanza_fmt = _INSTALL_FILE_STANZA_FMT_FEDORA40_DEBUGINFO
-
-            rpm_ctx.install_script_pieces.append(install_stanza_fmt.format(
+            rpm_ctx.install_script_pieces.append(_INSTALL_FILE_STANZA_FMT.format(
                 src.path,
                 abs_dest,
             ))
@@ -266,7 +251,7 @@ def _process_symlink(psi, origin_label, grouping_label, file_base, rpm_ctx):
         psi.attributes["mode"],
     ))
 
-def _process_dep(dep, rpm_ctx, debuginfo_type):
+def _process_dep(dep, rpm_ctx):
     # NOTE: This does not detect cases where directories are not named
     # consistently.  For example, all of these may collide in reality, but
     # won't be detected by the below:
@@ -290,7 +275,6 @@ def _process_dep(dep, rpm_ctx, debuginfo_type):
             None,  # group label
             _make_filetags(dep[PackageFilesInfo].attributes),  # file_base
             rpm_ctx,
-            debuginfo_type,
         )
 
     if PackageDirsInfo in dep:
@@ -321,7 +305,6 @@ def _process_dep(dep, rpm_ctx, debuginfo_type):
                 dep.label,
                 file_base,
                 rpm_ctx,
-                debuginfo_type,
             )
         for entry, origin in pfg_info.pkg_dirs:
             file_base = _make_filetags(entry.attributes, "%dir")
@@ -343,7 +326,7 @@ def _process_dep(dep, rpm_ctx, debuginfo_type):
                 rpm_ctx,
             )
 
-def _process_subrpm(ctx, rpm_name, rpm_info, rpm_ctx, debuginfo_type, effective_release = None):
+def _process_subrpm(ctx, rpm_name, rpm_info, rpm_ctx, effective_release = None):
     sub_rpm_ctx = struct(
         dest_check_map = {},
         install_script_pieces = [],
@@ -407,7 +390,7 @@ def _process_subrpm(ctx, rpm_name, rpm_info, rpm_ctx, debuginfo_type, effective_
         ]
 
         for dep in rpm_info.srcs:
-            _process_dep(dep, sub_rpm_ctx, debuginfo_type)
+            _process_dep(dep, sub_rpm_ctx)
 
         # rpmbuild will be unhappy if we have no files so we stick
         # default file mode in for that scenario
@@ -763,7 +746,7 @@ def _pkg_rpm_impl(ctx):
     # they aren't unnecessarily recreated.
 
     for dep in ctx.attr.srcs:
-        _process_dep(dep, rpm_ctx, debuginfo_type)
+        _process_dep(dep, rpm_ctx)
 
     #### subrpms
     if ctx.attr.subrpms:
@@ -774,7 +757,6 @@ def _pkg_rpm_impl(ctx):
                 rpm_name,
                 s[PackageSubRPMInfo],
                 rpm_ctx,
-                debuginfo_type,
                 effective_release = effective_release,
             ))
 
diff --git a/toolchains/rpm/rpmbuild.bzl b/toolchains/rpm/rpmbuild.bzl
index f07f819..532e1ad 100644
--- a/toolchains/rpm/rpmbuild.bzl
+++ b/toolchains/rpm/rpmbuild.bzl
@@ -62,7 +62,7 @@ rpmbuild_toolchain = rule(
             doc = """
             The underlying debuginfo configuration for the system rpmbuild.
 
-            One of `centos` (RPM < 4.18), `fedora` (RPM >= 4.18), and `none`
+            One of `centos` (RPM < 4.18 or >= 4.20), `fedora` (RPM 4.18 and 4.19), and `none`
             """,
             default = "none",
         ),
diff --git a/toolchains/rpm/rpmbuild_configure.bzl b/toolchains/rpm/rpmbuild_configure.bzl
index 3ab6a26..2b2ee7a 100644
--- a/toolchains/rpm/rpmbuild_configure.bzl
+++ b/toolchains/rpm/rpmbuild_configure.bzl
@@ -104,11 +104,12 @@ def _build_repo_for_rpmbuild_toolchain_impl(rctx):
         if len(version_parts) > 1 and version_parts[0].isdigit() and version_parts[1].isdigit():
             major = int(version_parts[0])
             minor = int(version_parts[1])
-            if major < 4 or (major == 4 and minor < 18):
-                debuginfo_type = DEBUGINFO_TYPE_CENTOS
-            else:
+            if major == 4 and minor in (18, 19):
                 # https://rpm.org/wiki/Releases/4.18.0: "Make %{buildsubdir} settable outside %setup"
+                # https://rpm.org/releases/4.20.0: "A per-package build directory that’s fully RPM-controlled is now used"
                 debuginfo_type = DEBUGINFO_TYPE_FEDORA
+            else:
+                debuginfo_type = DEBUGINFO_TYPE_CENTOS
         elif rctx.path(RELEASE_PATH).exists:
             rctx.watch(RELEASE_PATH)
             os_name, _ = _parse_release_info(rctx.read(RELEASE_PATH))
@@ -137,8 +138,8 @@ build_repo_for_rpmbuild_toolchain = repository_rule(
             The underlying debuginfo configuration for the system rpmbuild.
 
             One of:
-            - `centos` (RPM < 4.18),
-            - `fedora` (RPM >= 4.18),
+            - `centos` (RPM < 4.18 or >= 4.20),
+            - `fedora` (RPM 4.18 and 4.19),
             - `none`,
             - `default` (detects from `rpmbuild` version if available, otherwise looks up `/etc/os-release`)
             """,
