diff --git a/flex/internal/flex_action.bzl b/flex/internal/flex_action.bzl
index 66510e4..36f670a 100644
--- a/flex/internal/flex_action.bzl
+++ b/flex/internal/flex_action.bzl
@@ -95,9 +95,9 @@ def flex_action(ctx):
         src_ext = "c"
 
         # The Flex manual documents that `--header-file` and `--c++` are incompatible.
-        header = ctx.actions.declare_file("{}.h".format(ctx.attr.name))
-        args.add("--header-file=" + header.path)
-        outputs.append(header)
+        #header = ctx.actions.declare_file("{}.h".format(ctx.attr.name))
+        #args.add("--header-file=" + header.path)
+        #outputs.append(header)
     else:
         src_ext = "cc"
         args.add("--c++")
