From e07bee9290f5dde4456ecef98eed2ac7d2092d9c Mon Sep 17 00:00:00 2001
From: David du Colombier <djc@datadoghq.com>
Date: Wed, 19 Jul 2023 08:53:23 +0200
Subject: [PATCH 5/6] Fix leak of filename in oval_agent_new_session

---
 src/OVAL/oval_agent.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/OVAL/oval_agent.c b/src/OVAL/oval_agent.c
index 1fbed8ae6..a4ee34d12 100644
--- a/src/OVAL/oval_agent.c
+++ b/src/OVAL/oval_agent.c
@@ -112,6 +112,7 @@ oval_agent_session_t * oval_agent_new_session(struct oval_definition_model *mode
 	/* probe sysinfo */
 	ret = oval_probe_query_sysinfo(ag_sess->psess, &sysinfo);
 	if (ret != 0) {
+		free(ag_sess->filename);
 		oval_probe_session_destroy(ag_sess->psess);
 		oval_syschar_model_free(ag_sess->sys_model);
 		free(ag_sess);
-- 
2.34.1

