From 83592087efc58d72892d784cc709664ed91e81b3 Mon Sep 17 00:00:00 2001
From: David du Colombier <djc@datadoghq.com>
Date: Wed, 19 Jul 2023 08:46:55 +0200
Subject: [PATCH 6/6] Fix leak of item in probe_item_collect

---
 src/OVAL/probes/probe/icache.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/OVAL/probes/probe/icache.c b/src/OVAL/probes/probe/icache.c
index a397d35ec..04ddbfb9e 100644
--- a/src/OVAL/probes/probe/icache.c
+++ b/src/OVAL/probes/probe/icache.c
@@ -552,9 +552,11 @@ int probe_item_collect(struct probe_ctx *ctx, SEXP_t *item)
 	memcheck_ret = probe_cobj_memcheck(cobj_itemcnt, ctx->max_mem_ratio);
 	if (memcheck_ret == -1) {
 		dE("Failed to check available memory");
+		SEXP_free(item);
 		return -1;
 	}
 	if (memcheck_ret == 1) {
+		SEXP_free(item);
 
 		/*
 		 * Don't set the message again if the collected object is
-- 
2.34.1

