From 329587376a05aefffef864ebca98ea7bbaba4751 Mon Sep 17 00:00:00 2001
From: David du Colombier <djc@datadoghq.com>
Date: Mon, 20 Mar 2023 15:31:45 +0100
Subject: [PATCH 36/41] Fix leak of regex structure in oval_fts in error cases

---
 src/OVAL/probes/oval_fts.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/OVAL/probes/oval_fts.c b/src/OVAL/probes/oval_fts.c
index f9d0a0c1f..c96247b09 100644
--- a/src/OVAL/probes/oval_fts.c
+++ b/src/OVAL/probes/oval_fts.c
@@ -845,6 +845,7 @@ OVAL_FTS *oval_fts_open_prefixed(const char *prefix, SEXP_t *path, SEXP_t *filen
 			   errno, strerror(errno));
 		}
 		free((void *) paths[0]);
+		pcre_free(regex);
 		return NULL;
 	}
 
@@ -860,6 +861,7 @@ OVAL_FTS *oval_fts_open_prefixed(const char *prefix, SEXP_t *path, SEXP_t *filen
 	if (ofts->ofts_match_path_fts == NULL || errno != 0) {
 		dE("fts_open() failed, errno: %d \"%s\".", errno, strerror(errno));
 		OVAL_FTS_free(ofts);
+		pcre_free(regex);
 		return (NULL);
 	}
 
-- 
2.34.1

