--- a/src/OVAL/probes/unix/linux/rpminfo_probe.c
+++ b/src/OVAL/probes/unix/linux/rpminfo_probe.c
@@ -286,6 +286,7 @@ void *rpminfo_probe_init(void)
 {
 #ifdef RPM46_FOUND
 	rpmlogSetCallback(rpmErrorCb, NULL);
+	rpmSetVerbosity(RPMLOG_ERR);
 #endif
 	struct rpm_probe_global *g_rpm = malloc(sizeof(struct rpm_probe_global));
 	if (rpmReadConfigFiles ((const char *)NULL, (const char *)NULL) != 0) {
--- a/src/OVAL/probes/unix/linux/rpmverify_probe.c
+++ b/src/OVAL/probes/unix/linux/rpmverify_probe.c
@@ -229,6 +229,7 @@ void *rpmverify_probe_init(void)
 {
 #ifdef RPM46_FOUND
 	rpmlogSetCallback(rpmErrorCb, NULL);
+	rpmSetVerbosity(RPMLOG_ERR);
 #endif
         if (rpmReadConfigFiles ((const char *)NULL, (const char *)NULL) != 0) {
                 dD("rpmReadConfigFiles failed: %u, %s.", errno, strerror (errno));
--- a/src/OVAL/probes/unix/linux/rpmverifyfile_probe.c
+++ b/src/OVAL/probes/unix/linux/rpmverifyfile_probe.c
@@ -348,6 +348,7 @@ void *rpmverifyfile_probe_init(void)
 {
 #ifdef RPM46_FOUND
 	rpmlogSetCallback(rpmErrorCb, NULL);
+	rpmSetVerbosity(RPMLOG_ERR);
 #endif
 	if (rpmReadConfigFiles ((const char *)NULL, (const char *)NULL) != 0) {
 		dD("rpmReadConfigFiles failed: %u, %s.", errno, strerror (errno));
--- a/src/OVAL/probes/unix/linux/rpmverifypackage_probe.c
+++ b/src/OVAL/probes/unix/linux/rpmverifypackage_probe.c
@@ -338,6 +338,7 @@ void *rpmverifypackage_probe_init(void)
 
 #ifdef RPM46_FOUND
 	rpmlogSetCallback(rpmErrorCb, NULL);
+	rpmSetVerbosity(RPMLOG_ERR);
 #endif
 
 	if (CHROOT_IS_SET()) {
