--- a/src/OVAL/results/oval_resultTest.c
+++ b/src/OVAL/results/oval_resultTest.c
@@ -1144,6 +1144,13 @@ oval_result_t oval_result_test_eval(struct oval_result_test *rtest)
 	const char *comment = oval_test_get_comment(test);
 	dI("Evaluating %s test '%s': %s.", type, test_id, comment);
 
+	/* Skip systctl probe in offline mode */
+	if (getenv("OSCAP_PROBE_ROOT") != NULL && oval_test_get_subtype(test) == OVAL_UNIX_SYSCTL) {
+		dI("Skip %s probe in offline mode.", type);
+		rtest->result = OVAL_RESULT_TRUE;
+		return rtest->result;
+	}
+
 	if (rtest->result == OVAL_RESULT_NOT_EVALUATED) {
 		if (oval_test_get_subtype(oval_result_test_get_test(rtest)) != OVAL_INDEPENDENT_UNKNOWN) {
 			struct oval_string_map *tmp_map = oval_string_map_new();
