--- a/scripts/build/companion_libs/100-gmp.sh
+++ b/scripts/build/companion_libs/100-gmp.sh
@@ -127,7 +127,7 @@
     # CC_FOR_BUILD and CPP_FOR_BUILD.
     CT_DoExecLog CFG                                \
     CC_FOR_BUILD="${CT_BUILD}-gcc"                  \
-    CPP_FOR_BUILD="{CT_BUILD}-cpp"                  \
+    CPP_FOR_BUILD="${CT_BUILD}-cpp"                  \
     CC="${host}-gcc"                                \
     CFLAGS="${cflags} -fexceptions"                 \
     LDFLAGS="${ldflags}"                            \
