--- a/pcap/BUILD.bazel
+++ b/pcap/BUILD.bazel
@@ -6,6 +6,10 @@
         "pcap_unix.go",
         "pcap_windows.go",
     ],
+    cdeps = select({
+        "@io_bazel_rules_go//go/platform:linux": ["@libpcap//:pcap"],  # keep
+        "//conditions:default": [],
+    }),
     cgo = True,
     clinkopts = select({
         "@io_bazel_rules_go//go/platform:android": [
@@ -21,7 +25,6 @@
             "-lpcap",
         ],
         "@io_bazel_rules_go//go/platform:linux": [
-            "-lpcap",
         ],
         "@io_bazel_rules_go//go/platform:netbsd": [
             "-lpcap",
