https://github.com/Supervisor/supervisor/pull/1578
diff --git a/setup.py b/setup.py
index ac4373c..d6449c9 100644
--- a/setup.py
+++ b/setup.py
@@ -22,8 +22,10 @@ if py_version < (2, 7):
 elif (3, 0) < py_version < (3, 4):
     raise RuntimeError('On Python 3, Supervisor requires Python 3.4 or later')

-# pkg_resource is used in several places
-requires = ["setuptools"]
+requires = [
+    "importlib-metadata; python_version < '3.8'",
+    "importlib-resources; python_version < '3.9'",
+]
 tests_require = []
 if py_version < (3, 3):
     tests_require.append('mock<4.0.0.dev0')
