https://github.com/Supervisor/supervisor/pull/1578
diff --git a/supervisor/confecho.py b/supervisor/confecho.py
index c137b75..2bcafe2 100644
--- a/supervisor/confecho.py
+++ b/supervisor/confecho.py
@@ -1,7 +1,12 @@
-import pkg_resources
 import sys
 from supervisor.compat import as_string

+if sys.version_info >= (3, 9):
+    from importlib.resources import files as resource_files
+else:
+    from importlib_resources import files as resource_files
+
+
 def main(out=sys.stdout):
-    config = pkg_resources.resource_string(__name__, 'skel/sample.conf')
+    config = resource_files(__package__).joinpath('skel/sample.conf').read_text(encoding='utf-8')
     out.write(as_string(config))
