diff --git a/overlays/toolchain/BUILD.args-msvc.tpl b/overlays/toolchain/BUILD.args-msvc.tpl
index 7cb5304..77c285b 100644
--- a/overlays/toolchain/BUILD.args-msvc.tpl
+++ b/overlays/toolchain/BUILD.args-msvc.tpl
@@ -136,7 +136,7 @@ cc_args(
     name = "param_file_args",
     actions = [
         "@rules_cc//cc/toolchains/actions:link_actions",
-        "@rules_cc//cc/toolchains/actions:ar_actions",
+        "@rules_cc//cc/toolchains/actions:cpp_link_static_library",
     ],
     args = ["@{param_file}"],
     format = {"param_file": "@rules_cc//cc/toolchains/variables:linker_param_file"},
@@ -145,7 +145,7 @@ cc_args(
 
 cc_args(
     name = "archiver_output",
-    actions = ["@rules_cc//cc/toolchains/actions:ar_actions"],
+    actions = ["@rules_cc//cc/toolchains/actions:cpp_link_static_library"],
     args = ["/OUT:{output_execpath}"],
     format = {"output_execpath": "@rules_cc//cc/toolchains/variables:output_execpath"},
     requires_not_none = "@rules_cc//cc/toolchains/variables:output_execpath",
@@ -153,7 +153,7 @@ cc_args(
 
 cc_args(
     name = "archiver_input",
-    actions = ["@rules_cc//cc/toolchains/actions:ar_actions"],
+    actions = ["@rules_cc//cc/toolchains/actions:cpp_link_static_library"],
     nested = [":archiver_libraries_to_link_iterate"],
     requires_not_none = "@rules_cc//cc/toolchains/variables:libraries_to_link",
 )
