diff --git a/text/exec.go b/text/exec.go
index b6febc5..92ce842 100644
--- a/pkg/template/text/exec.go
+++ b/pkg/template/text/exec.go
@@ -643,15 +643,6 @@ func (s *state) evalField(dot reflect.Value, fieldName string, node parse.Node,
 		return zero
 	}
 
-	// Unless it's an interface, need to get to a value of type *T to guarantee
-	// we see all methods of T and *T.
-	ptr := receiver
-	if ptr.Kind() != reflect.Interface && ptr.Kind() != reflect.Pointer && ptr.CanAddr() {
-		ptr = ptr.Addr()
-	}
-	if method := ptr.MethodByName(fieldName); method.IsValid() {
-		return s.evalCall(dot, method, false, node, fieldName, args, final)
-	}
 	hasArgs := len(args) > 1 || !isMissing(final)
 	// It's not a method; must be a field of a struct or an element of a map.
 	switch receiver.Kind() {
