--- pkg/proto/pbgo/trace/span_gen.go.after_first_patch	2025-09-09 16:34:07
+++ pkg/proto/pbgo/trace/span_gen.go	2025-09-09 16:34:17
@@ -691,7 +691,7 @@
 				err = msgp.WrapError(err, "Meta")
 				return
 			}
-			if z.Meta == nil {
+			if z.Meta == nil && zb0002 > 0 {
 				z.Meta = make(map[string]string, zb0002)
 			} else if len(z.Meta) > 0 {
 				clear(z.Meta)
@@ -724,7 +724,7 @@
 				err = msgp.WrapError(err, "Metrics")
 				return
 			}
-			if z.Metrics == nil {
+			if z.Metrics == nil && zb0003 > 0 {
 				z.Metrics = make(map[string]float64, zb0003)
 			} else if len(z.Metrics) > 0 {
 				clear(z.Metrics)
