--- CMakeLists.txt	2020-12-18 19:57:38.000000000 +0000
+++ CMakeLists.txt	2022-02-09 22:07:27.926584000 +0000
@@ -68,6 +68,9 @@ if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "Build type (default Debug)" FORCE)
 endif()

+add_compile_options("-Wl,--wrap=exp -Wl,--wrap=log -Wl,--wrap=log2 -Wl,--wrap=log2f")
+link_libraries("-Wl,--wrap=exp -Wl,--wrap=log -Wl,--wrap=log2 -Wl,--wrap=log2f")
+
 # Side-by-side subprojects layout: automatically set the
 # LLVM_EXTERNAL_${project}_SOURCE_DIR using LLVM_ALL_PROJECTS
 # This allows an easy way of setting up a build directory for llvm and another
